### ISSUE TITLE
Hyperparameter optimization and TensorFlow

### ISSUE THREAD

**Issue by Carlos-Henreis** [tensorflow/tensorflow#18525]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution:** Linux-Ubuntu (17.10)
- **TensorFlow installed from**: Binary
- **TensorFlow version**:  1.5.0
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem
I know that this question has already been addressed here and in Stack Overflow, but until the moment I do not see many evolutions on the subject.
Saw this approach here [A Scikit-learn compatible Deep Neural Network built with TensorFlow](https://medium.com/@williamkoehrsen/deep-neural-network-classifier-32c12ff46b6c),, that is, using TensorFlow (model build) and Scikit-learn to make the model's Hyperparameter Tuning.
So far this approach is the most correct?
Really wanted to be able to contribute and work with the optimization of hyperparameters along with the TensorFlow models.
I find this technique very important in Deep Learning...
thanks in advance

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by bhack** [comment]
Take a look to the Kubeflow related project: https://github.com/kubeflow/hp-tuning

**Comment by Carlos-Henreis** [comment]
Hello,
I've not had any feedback nor here much less in stackoverflow ...
I've seen several alternatives, one of which I have described here before, and another in this repository [here](https://github.com/charuj/multigraph_lstm_hyperparamsearch)

Consequently I think the second most attractive alternative.

In my final paper (graduation in computer science) I approached the second alternative applying the automatic selection of hyperparameters in DNNClassfier ...

It is still crawling more I think it is possible to create an Estimator following this proposal ... I hope to spread my studies soon ...

I just need to mature better ... For now is can create an example of how to use the optimization of hyperparameters along with Tensorflow ... Reinforcing I think this is very interesting ... And I see that this is still a limitation of Tensorflow

**Comment by bhack** [comment]
Katib is trying to replicate Google vizier concepts in an opensource impl:
https://github.com/kubeflow/katib

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by dynamicwebpaige** [comment]
Closing, as this capability is added in [Keras Tuner](https://github.com/keras-team/keras-tuner).
